src/test/java/Overview/DesignTest.java

```diff
@@ -0,0 +1,305 @@
+     * aspects of the design flow.
+     */
+    @Test
+    @DisplayName("Design Overview Test")
+    public void testDesignOverview() {
+        // create the top level design object
+        CellDesign design = new CellDesign("TestDesignOverview", device.getPartName());
+        assertEquals("TestDesignOverview", design.getName(), "Design name doesn't match constructor");
+
+        // find a LUT cell to use for this design
+        LibraryCell lutlibcell = libCells.get("LUT3");
+        assertEquals("LUT3", lutlibcell.getName(), "LUT LibraryCell name doesn't match");
+        assertEquals(new Integer(3), lutlibcell.getNumLutInputs(), "Incorrect input count for LUT LibraryCell");
+
+        // create a LUT to use for the sum logic
+        Cell addcell = design.addCell(new Cell("addcell", lutlibcell));
+        assertEquals("addcell", addcell.getName(), "LUT cell name doesn't match constructor");
+        // Program the LUT
+        addcell.getProperties().update("INIT", PropertyType.DESIGN, "8'b01101001");
+        /* truth table for sum logic
+          a b c | o
+          0 0 0 | 0
+          0 0 1 | 1
+          0 1 0 | 1
+          0 1 1 | 0
+          1 0 0 | 1
+          1 0 1 | 0
+          1 1 0 | 0
+          1 1 1 | 1
+        */
+        testLUT3(addcell);
+
+        // create a LUT cell for the carry logic
+        Cell carrycell = design.addCell(new Cell("carrycell", lutlibcell));
+        assertEquals("carrycell", carrycell.getName(), "LUT cell name doesn't match constructor");
+        // Program the LUT
+        carrycell.getProperties().update("INIT", PropertyType.DESIGN, "8'b00010111");
+        /* truth table for carry logic
+          a b c | o
+          0 0 0 | 0
+          0 0 1 | 0
+          0 1 0 | 0
+          0 1 1 | 1
+          1 0 0 | 0
+          1 0 1 | 1
+          1 1 0 | 1
+          1 1 1 | 1
+        */
+        testLUT3(carrycell);
+
+        // find a flip flop to use fo this design
+        LibraryCell fflibcell = libCells.get("FDRE");
+
+        // Create a flip flop cell to register the sum output
+        Cell sumreg = design.addCell(new Cell("sumreg", fflibcell));
+        assertEquals("sumreg", sumreg.getName(), "Flip Flop cell name doesn't match constructor");
+        sumreg.getProperties().update("INIT", PropertyType.DESIGN, "INIT0");
+        sumreg.getProperties().update("SR", PropertyType.DESIGN, "SRLOW");
+        testFF(sumreg);
+
+        // Create a flip flop cell to register the carry output
+        Cell coutreg = design.addCell(new Cell("coutreg", fflibcell));
+        assertEquals("coutreg", coutreg.getName(), "Flip Flop cell name doesn't match constructor");
+        coutreg.getProperties().update("INIT", PropertyType.DESIGN, "INIT0");
+        coutreg.getProperties().update("SR", PropertyType.DESIGN, "SRLOW");
+        testFF(coutreg);
+
+        // create and test the input ports
+        Cell aport = design.addCell(new Cell("aport", libCells.get("IPORT")));
```

Comment:
Because the ports are referenced later in the design (to connect them to the nets), this would just make it less clean because I would have to indirectly find the ports to retrieve the pins.